docs/language.rst

```diff
@@ -864,3 +864,76 @@ Plug-ins
 For more complex operations, python plugins can be used. Plugins are exposed in the Inmanta language as function calls, such as the template function call. A template
 accepts parameters and returns a value that it computed out of the variables. Each module that is included can also provide plug-ins. These plug-ins are accessible within the namespace of the
 module. The :ref:`module-plugins` section of the module guide provides more details about how to write a plugin.
+
+
+.. _language_unknowns:
+Unknowns
+========
+
+Wherever the configuration model interacts with the outside world (e.g. to fetch external values) :term:`unknown` values
+may be present. These unknowns are propagated through the model to finally end up in the resources that require these unknown
```

Comment:
I would add a sentence to say what they are. 

```suggestion
may be present. These unknowns represent value we don't know yet, such as the IP address of a machine that hasn't been created yet. Because the compiler can handle unknowns, developer can writes models as if all information is present up front, even when this is not the case. These unknowns are propagated through the model to finally end up in the resources that require these unknown
```